stdlib/functools.pyi

```diff
@@ -2,12 +2,15 @@ import sys
 import types
 from _typeshed import SupportsAllComparisons, SupportsItems
 from collections.abc import Callable, Hashable, Iterable, Sequence, Sized
-from typing import Any, Generic, Literal, NamedTuple, TypedDict, TypeVar, final, overload
+from typing import Any, Generic, Literal, NamedTuple, Protocol, TypedDict, TypeVar, final, overload
 from typing_extensions import ParamSpec, Self, TypeAlias
 
 if sys.version_info >= (3, 9):
     from types import GenericAlias
 
+if sys.version_info >= (3, 10):
+    from typing import Concatenate
+
```

Comment:
```suggestion
from typing_extensions import Concatenate, ParamSpec, Self, TypeAlias

if sys.version_info >= (3, 9):
    from types import GenericAlias

```

See [CONTRIBUTING](https://github.com/python/typeshed/blob/main/CONTRIBUTING.md#supported-type-system-features) for which type system features are supported by typeshed and which ones need to be imported from `typing_extensions` rather than `typing`.

Reply:
Thank you, I've been thinking about this (and actually reading documentation instead of blindly trying to get stuff to work). My thought at the moment is as you've suggested use the `typing_extensions`, also change back to `_lru_cache_wrapper` as a base class then have the descriptors as `typing.type_check_only` subclasses. I think the super/subclass issue is a correct error. And I need to add back into the `stubtest_allowlist` that the classes I've created are valid/to be ignored.